arangod/Aql/EnumerateCollectionExecutor.cpp

```diff
@@ -57,7 +56,7 @@ EnumerateCollectionExecutorInfos::EnumerateCollectionExecutorInfos(
     RegisterId outputRegister, aql::QueryContext& query,
     Collection const* collection, Variable const* outVariable, bool produceResult,
     Expression* filter, arangodb::aql::Projections projections,
-    bool random, bool count)
+    bool random, bool count, ReadOwnWrites readOwnWrites)
```

Comment:
This would introduce an indirection that is not needed. Currently the values is either passed as register (unlikely, because there are already a lot of parameter) or directly on the stack. Having it as `const&` (or as r-value ref `&&` would require the compiler to write the value to memory somewhere, in order to pass a pointer to the function as parameter, via register or on the stack. Please correct me if I am wrong.

Reply:
@maierlars is correct. Since the type's size is less than pointer size, it is preferable to pass-by-value instead of by-ref.